### PR TITLE
changes made to the csv file

### PR DESCRIPTION
**Pull request by jsaltane123**
Changes were made to the .csv and associated discovery file, as there were a few inconsistencies, typos and missing variables.
The finding were first raised with the team (Ben, Amelia and Daniel) and then further discussed during a call.
I am attaching a file with detailed description of findings per each ADaM datatable.

[admiraldiscovery_issue.docx](https://github.com/user-attachments/files/16190624/admiraldiscovery_issue.docx)

Note that the suite was built locally to check the updated table.

### CODE REVIEW

File: inst/admiral-lookup-book.csv [hunk 1]
```diff
@@ -1,44 +1,81 @@
...
-ADEG,BDS,PARAMCD:QTCBR,Derive a parameter for corrected QT using Bazett's,admiral,derive_param_qtc,https://pharmaverse.github.io/admiral/reference/derive_param_qtc.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adeg.R
-ADEG,BDS,PARAMCD:QTCFR,Derive a parameter for corrected QT using Fridericia's,admiral,derive_param_qtc,https://pharmaverse.github.io/admiral/reference/derive_param_qtc.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adeg.R
-ADEG,BDS,PARAMCD:QTLCR,Derive a parameter for corrected QT using Sagie's ,admiral,derive_param_qtc,https://pharmaverse.github.io/admiral/reference/derive_param_qtc.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adeg.R
+ADEG,BDS,ADY,Derive analysis relative day,admiral,derive_vars_dy,https://pharmaverse.github.io/admiral/reference/derive_vars_dy.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adeg.R
```
Comment (bms63) [inline inst/admiral-lookup-book.csv:71]: This is missing day

File: inst/admiral-lookup-book.csv [hunk 2]
```diff
@@ -1,44 +1,81 @@
...
+ADEG,BDS,PARAM,Derive a list of parameter codes,admiral,derive_vars_merged_lookup,https://pharmaverse.github.io/admiral/reference/derive_vars_merged_lookup.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adeg.R
```
Comment (bms63) [inline inst/admiral-lookup-book.csv:72]: how come this become "list of parameter"

File: inst/admiral-lookup-book.csv [hunk 3]
```diff
@@ -48,26 +85,67 @@ ADEG,BDS,ABLFL,Derive analysis baseline flag,admiral,restrict_derivation,https:/
...
-ADEX,OCCDS,TRTSDT,Derive treatment start date ,admiral,derive_vars_merged,https://pharmaverse.github.io/admiral/reference/derive_vars_merged.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adex.R
-ADEX,OCCDS,ASTDTM,Derive analysis start datetime,admiral,derive_vars_dtm,https://pharmaverse.github.io/admiral/reference/derive_vars_dtm.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adex.R
-ADEX,OCCDS,AENDTM,Derive analysis end datetime,admiral,derive_vars_dtm,https://pharmaverse.github.io/admiral/reference/derive_vars_dtm.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adex.R
+ADEX,OCCDS,TRTSDT,Derive treatment start date,admiral,derive_vars_merged,https://pharmaverse.github.io/admiral/reference/derive_vars_merged.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adex.R
+ADEX,OCCDS,AENDTM,Derive analysis end datetime and date/time imput. flags,admiral,derive_vars_dtm,https://pharmaverse.github.io/admiral/reference/derive_vars_dtm.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adex.R
```
Comment (bms63) [inline inst/admiral-lookup-book.csv:89]:
```suggestion
ADEX,OCCDS,AENDTM,Derive analysis end datetime and date/time impute. flags,admiral,derive_vars_dtm,https://pharmaverse.github.io/admiral/reference/derive_vars_dtm.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adex.R
```

File: inst/admiral-lookup-book.csv (hunk 1 above)
Comment (bms63) [reply inst/admiral-lookup-book.csv:71]: oh i see it is on next line - please disregard

File: inst/admiral-lookup-book.csv [hunk 4]
```diff
@@ -48,26 +85,67 @@ ADEG,BDS,ABLFL,Derive analysis baseline flag,admiral,restrict_derivation,https:/
...
+ADFACE,ADSL,ANL01FL,Derive flags for the maximum record per subject per event per vaccination,admiralvaccine,derive_vars_max_flag,https://pharmaverse.github.io/admiralvaccine/reference/derive_vars_max_flag.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
+ADFACE,ADSL,ANL02FL,Derive flags for the maximum record per subject per event for overall,admiralvaccine,derive_vars_max_flag,https://pharmaverse.github.io/admiralvaccine/reference/derive_vars_max_flag.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
+ADFACE,ADSL,EVENTFL,Derive flags the record if at least one of the event occurred within the observation period,admiralvaccine,derive_vars_event_flag,https://pharmaverse.github.io/admiralvaccine/reference/derive_vars_event_flag.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
```
Comment (bms63) [inline inst/admiral-lookup-book.csv:117]:
```suggestion
ADFACE,ADSL,EVENTFL,Derive flags for the record if at least one of the event occurred within the observation period,admiralvaccine,derive_vars_event_flag,https://pharmaverse.github.io/admiralvaccine/reference/derive_vars_event_flag.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
```

File: inst/admiral-lookup-book.csv [hunk 5]
```diff
@@ -48,26 +85,67 @@ ADEG,BDS,ABLFL,Derive analysis baseline flag,admiral,restrict_derivation,https:/
...
+ADFACE,ADSL,EVENTDFL,Derive flags the record if the event is occurred,admiralvaccine,derive_vars_event_flag,https://pharmaverse.github.io/admiralvaccine/reference/derive_vars_event_flag.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
```
Comment (bms63) [inline inst/admiral-lookup-book.csv:118]:
```suggestion
ADFACE,ADSL,EVENTDFL,Derive flags for the record if the event is occurred,admiralvaccine,derive_vars_event_flag,https://pharmaverse.github.io/admiralvaccine/reference/derive_vars_event_flag.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
```

File: inst/admiral-lookup-book.csv [hunk 6]
```diff
@@ -95,245 +173,237 @@ ADLB,BDS,ASEQ,Derive sequential number for observations within each by group,adm
...
+ADOE,ADSL,ABLFL,Derive analysis baseline flag,admiral,restrict_derivation,https://pharmaverse.github.io/admiral/reference/restrict_derivation.html,Template Example,https://github.com/pharmaverse/admiralophtha/blob/main/inst/templates/ad_adoe.R
+ADOE,ADSL,ADT,Derive analysis date,admiral,derive_vars_dt,https://pharmaverse.github.io/admiral/reference/derive_vars_dt.html,Template Example,https://github.com/pharmaverse/admiralophtha/blob/main/inst/templates/ad_adoe.R
+ADOE,ADSL,ADY,Derive analysis relative day,admiral,derive_vars_dy,https://pharmaverse.github.io/admiral/reference/derive_vars_dy.html,Template Example,https://github.com/pharmaverse/admiralophtha/blob/main/inst/templates/ad_adoe.R
+ADOE,ADSL,ANL01FL,Derive flag last result within a visit and timepoint for baseline and post-baseline records,admiral,restrict_derivation,https://pharmaverse.github.io/admiral/reference/restrict_derivation.html,Template Example,https://github.com/pharmaverse/admiralophtha/blob/main/inst/templates/ad_adoe.R
```
Comment (bms63) [inline inst/admiral-lookup-book.csv:200]:
```suggestion
ADOE,ADSL,ANL01FL,Derive flag for last result within a visit and timepoint for baseline and post-baseline records,admiral,restrict_derivation,https://pharmaverse.github.io/admiral/reference/restrict_derivation.html,Template Example,https://github.com/pharmaverse/admiralophtha/blob/main/inst/templates/ad_adoe.R
```

Comment (bms63) [reply inst/admiral-lookup-book.csv:200]: Unsure on this one - others might need to be looked at as well

**Review by bms63 — CHANGES_REQUESTED** [review]
Wow! This is very thorough!  Thank you for doing this update.  @ddsjoberg is there a standard convention you all have somewhere or something we should make?  Like a simple wiki page referenced at top of file?

File: inst/admiral-lookup-book.csv (hunk 2 above)
Comment (jsaltane123) [reply inst/admiral-lookup-book.csv:72]: thanks for spotting, missed this one. This should not be PARAM, but PARAMCD. 
image

File: inst/admiral-lookup-book.csv (hunk 3 above)
Comment (jsaltane123) [reply inst/admiral-lookup-book.csv:89]: Apologies, missed these ones. So during our conversation it was if, let's say, AENDTM is derived in ADEX and you get imputations (AENDTF and AEDNTMF), we capture this as three variables:
1) AENDTM  - Derive analysis end datetime
2) AENDTF - Derive analysis end date imputation flag
3) AENTMF - Derive analysis end time imputation flag

I have added imputation flags as new variables, but have not adjusted the datetime variable descriptions correctly.

File: inst/admiral-lookup-book.csv [hunk 7]
```diff
@@ -95,245 +173,237 @@ ADLB,BDS,ASEQ,Derive sequential number for observations within each by group,adm
...
-ADPC,OCCDS,PCRFTTM,Derive reference dose time,admiral,derive_vars_dtm_to_tm,https://pharmaverse.github.io/admiral/reference/derive_vars_dtm_to_tm.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adpc.R
 ADPC,OCCDS,BASE,Derive BASE,admiral,derive_var_base,https://pharmaverse.github.io/admiral/reference/derive_var_base.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adpc.R
 ADPC,OCCDS,CHG,Derive change from baseline,admiral,derive_var_chg,https://pharmaverse.github.io/admiral/reference/derive_var_chg.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adpc.R
-ADPC,OCCDS,ASEQ,Derive sequential number for observations within each by group,admiral,derive_var_obs_number,https://pharmaverse.github.io/admiral/reference/derive_var_obs_number.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adpc.R
+ADPC,OCCDS,EXDOSE_next,Derive next dose taken,admiral,derive_vars_joined,https://pharmaverse.github.io/admiral/reference/derive_vars_joined.html,Template Example,https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adpc.R
```
[the remark this answers is not in the document]
Comment (jsaltane123) [reply inst/admiral-lookup-book.csv:240]: It is a variable which was just moved to another row, this was not added. But it does look like SDTM EX variable

File: inst/admiral-lookup-book.csv [hunk 8]
```diff
@@ -48,26 +85,67 @@ ADEG,BDS,ABLFL,Derive analysis baseline flag,admiral,restrict_derivation,https:/
...
+ADFACE,ADSL,APEREDT,Derive period end date,admiral,create_period_dataset,https://pharmaverse.github.io/admiral/reference/create_period_dataset.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
+ADFACE,ADSL,TRTA,Derive actual treatment,admiral,create_period_dataset,https://pharmaverse.github.io/admiral/reference/create_period_dataset.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
+ADFACE,ADSL,TRTP,Derive planned treatment,admiral,create_period_dataset,https://pharmaverse.github.io/admiral/reference/create_period_dataset.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
+ADFACE,ADSL,"FATESTCD=""SEV""","Derive  severity records from diameter for redness, swelling, etc",admiralvaccine,derive_diam_to_sev_records,https://pharmaverse.github.io/admiralvaccine/reference/derive_diam_to_sev_records.html,Template Example,https://github.com/pharmaverse/admiralvaccine/blob/main/inst/templates/ad_adface.R
```
[the remark this answers is not in the document]
Comment (jsaltane123) [reply inst/admiral-lookup-book.csv:111]: It does look like too many quotes here, but when I built the site, it looks ok: 
image

**Comment by jsaltane123** [comment]
Changes made on 15-Jul-2024
'[additional changes following review](https://github.com/pharmaverse/admiraldiscovery/pull/50/commits/9379f65cde112adb2cfc0cc5d81f84251cb0b9ba)' commit

Addressing comments (thanks a lot for the review, @bms63)
-	ADEG dataset – PARAM was renamed to PARAMCD (as description states ‘ Derive a list of parameter codes’
-	As imputation flags were added as separate variables, the description of corresponding date/time/datetime variables should not mention imputation flags. The following variables were adjusted: 
            o	ADAE – ASTDTM
            o	ADEX – ASTDTM and AENDTM
            o	ADPC – ASTDTM
            o	ADSL – DTHDT

Additional items:
-	ADPPK – ASTDT and AENDT – description was slightly changed:  from ‘…from date/time’ to ‘…from datetimes’. 
-	Description for PARAM variable was changed to be consistent across the datasets – ‘Derive a list of parameters’ (similar to ADOE and ADIS) 
-	Description for ‘PARAMCD’ was changed to be consistent across the datasets – ‘Derive a list of parameter codes’ (similar to ADVS)

**Comment by bms63** [comment]
@athenamelia do you mind reviewing.  I believe @ddsjoberg is away this week

File: inst/admiral-lookup-book.csv (hunk 3 above)
Comment (athenamelia) [reply inst/admiral-lookup-book.csv]: Screenshot 2024-07-15 at 11 10 21 AM

@jsaltane123 thanks Julija so much for the work. I'm going through the spreadsheet now. Just wanted to follow up with this comment about AENDTM variables across ADAE, ADCM, ADEX, ADPC, and ADPPK

**Review by athenamelia — COMMENTED** [review]
Thanks @jsaltane123 for the updates! This is very thorough. I looked at the spreadsheet and the additional updates to address the comments above -- it looks great!

Minor thing: 
- ADMH with variable label "Derive query variables": variable name is NA. Should we add one of SMQzzNAM, SMQzzCD, SMQzzSC, SMQzzSCN, and CQzzNAM as an example for query variable?

**Comment by athenamelia** [comment]
Inconsistency for ASTDT and AENDT description labels across ADaM. "Derive analysis start/end date" vs "Derive... from datetimes".

**Comment by jsaltane123** [comment]
Changes made on 16-Jul-2024
[commit](https://github.com/pharmaverse/admiraldiscovery/pull/50/commits/c241822c9d4eb9f2c2ccbc22a3344f2574fc17c2)

Addressing comments (thanks a lot for the review, @athenamelia )

ADMH dataset:
- empty variable with description ' Derive query variables' was removed;
- rows for SMQ02NAM, SMQ02CD, SMQ02SC, SMQ02SCN, CQ04NAM, CQ04CD were added;

All datasets:
- ASTDT description standardized to 'Derive analysis start date'
- AENDT description standardized to 'Derive analysis end date'
- ADT description standardized to  'Derive analysis date'
- ADTM description standardized to 'Derive analysis datetime'
- ATM - 'Derive analysis time'
- ATMF - 'Derive analysis time imputation flag'
- TRT01P - 'Derive planned treatment for period 1' (there was a mistake in ADVFQ)

**Comment by athenamelia** [comment]
This is great work, Julija @jsaltane123 ! Thanks for your contribution. @ddsjoberg @bms63 I don't have further comments from my end.

**Comment by bms63** [comment]
@jsaltane123 can I add you to the admiral community team.  I ping folks for help on issues in different admiral projects

**Review by bms63 — APPROVED** [review]
It looks great to me.  Let's wait for @ddsjoberg to come back to give final seal of approval.

Also @jsaltane123 - we just had an admiral retrospectives meeting today and a new working group will be starting in later August that will look at ADAM IG Guide and what is missing in admiral.  I was thinking of the TREATCOM - Treatment Compliance variable discussion that we had last week when proposing the working group.  Want to get invovled!?!

**Comment by jsaltane123** [comment]
@bms63 , thanks a lot for considering me for the Admiral community team. I would be happy to join and contribute to the various projects! There are other Ascent colleagues who are also keen to contribute and will introduce them at a later stage. 
The new working group focused on the ADAM IG Guide sounds particularly interesting, and I'm excited about the opportunity to help address any gaps in Admiral! :)

**Comment by ddsjoberg** [comment]
HI All, I am back! Thanks for the amazing work on this PR!

I think everything looks in order. But just to be sure, can we resolve the outstanding comments? When we resolve, can we make a note whether that suggested change was made, not made (and why)?

File: inst/admiral-lookup-book.csv (hunk 7 above)
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply inst/admiral-lookup-book.csv:239]: If this is not an ADaM variable, should we not just remove? Thanks!

File: inst/admiral-lookup-book.csv (hunk 6 above)
Comment (jsaltane123) [reply inst/admiral-lookup-book.csv:199]: @bms63 @ddsjoberg @athenamelia Could you please advise me on this one? Is the description correct or shall it be adjusted?

File: inst/admiral-lookup-book.csv (hunk 7 above)
[the remark this answers is not in the document]
Comment (jsaltane123) [reply inst/admiral-lookup-book.csv:239]: Sure, will remove it

**Comment by ddsjoberg** [comment]
@jsaltane123 thank you again for your wonderful contribution! 🕺🏼 

We just need one last step. Can you run `devtools::document()` and push the update? I tried to do it myself, but I am blocked from modifying your branch.

**Comment by ddsjoberg** [comment]
On second thought @jsaltane123 , I can go ahead and merge this now, then push the redocs separately. You've already done so much for this PR and it's VERY MUCH appreciated!

**Comment by jsaltane123** [comment]
@bms63 , @athenamelia and @ddsjoberg thank you so much for the opportunity to contribute and for your guidance :) I look forward to making further contributions!

**Pull request merged**